pkg/eks/tasks.go

```diff
@@ -229,12 +277,27 @@ func (c *ClusterProvider) appendCreateTasksForIAMServiceAccounts(cfg *api.Cluste
 		},
 	}
 
+	serviceAccounts := append([]*api.ClusterIAMServiceAccount{}, cfg.IAM.ServiceAccounts...)
+	if api.IsEnabled(cfg.IAM.WithOIDC) {
+		serviceAccounts = append(serviceAccounts, &api.ClusterIAMServiceAccount{
+			ClusterIAMMeta: api.ClusterIAMMeta{Name: "aws-node",
+				Namespace: "kube-system",
+			},
+			AttachPolicyARNs: []string{fmt.Sprintf("arn:aws:iam::aws:policy/%s", iamPolicyAmazonEKSCNIPolicy)},
```

Comment:
Is this intentional that aws partition isn't used in the arn?
Will this work correctly with aws-cn/aws-us-gov?
I'm not familiar with the code so I don't know if it will be transformed later.

Reply:
Good catch, will be fixed in the RC on thursday.